contrib/server/operations/pbench-upload-results.py

```diff
@@ -0,0 +1,270 @@
+record the hostname, and a marker that it was "migrated" by this utility.
+"""
+from argparse import ArgumentParser
+import datetime
+from http import HTTPStatus
+from io import TextIOWrapper
+from pathlib import Path
+import socket
+import sys
+from typing import Optional
+
+from dateutil import parser as date_parser
+import requests
+from requests import Response
+
+
+def get_md5(tarball: Path) -> str:
+    """Read the tarball MD5 from the tarball's companion file
+
+    Args:
+        tarball: Path to a tarball file with a {name}.md5 companion
+
+    Returns:
+        The tarball's MD5 value
+    """
+    md5_file = Path(f"{str(tarball)}.md5")
+    return md5_file.read_text().split()[0]
+
+
+def upload(
+    server: str, token: str, tarball: Path, metadata: Optional[list[str]] = None
+) -> Response:
+    query_parameters = {}
+
+    md5 = get_md5(tarball)
+    query_parameters["access"] = "public"
+    dataset = tarball.name
+    satellite, _ = tarball.parent.name.split("::")
+    meta = [
+        f"global.server.legacy.migrated:{datetime.datetime.now(tz=datetime.timezone.utc):%Y-%m-%dT%H:%M}"
+    ]
+    if satellite:
+        meta.append(f"server.origin:{satellite}")
+    if metadata:
+        meta.extend(metadata)
+    query_parameters["metadata"] = meta
+    headers = {
+        "Content-MD5": md5,
+        "content-type": "application/octet-stream",
+        "authorization": f"bearer {token}",
+    }
+
+    with tarball.open("rb") as f:
+        return requests.put(
+            f"{server}/api/v1/upload/{dataset}",
+            headers=headers,
+            params=query_parameters,
+            data=f,
+        )
+
+
+def main() -> int:
+    prog = Path(sys.argv[0])
+    parser = ArgumentParser(prog=prog.name, description="Upload tarballs")
+    parser.add_argument("server", help="Specify the Pbench Server address")
+    parser.add_argument(
+        "tarball", type=Path, help="Specify a tarball or directory path"
+    )
+    parser.add_argument("-b", "--before", help="Select tarballs older than this")
+    parser.add_argument(
+        "-c",
+        "--checkpoint",
+        type=Path,
+        dest="checkpoint",
+        help="Checkpoint file for restart",
+    )
+    parser.add_argument(
+        "-m",
+        "--metadata",
+        action="append",
+        dest="metadata",
+        help="Set metadata on dataset upload",
+    )
+    parser.add_argument(
+        "-p",
+        "--preview",
+        action="store_true",
+        help="Report actions but make no changes",
+    )
+    parser.add_argument(
+        "-s", "--since", action="store", help="Select tarballs no older than this"
+    )
+    parser.add_argument(
+        "-t", "--token", action="store", dest="token", help="Pbench Server API token"
+    )
+    parser.add_argument(
+        "--verify", "-v", dest="verify", action="store_true", help="Show progress"
+    )
+    parsed = parser.parse_args()
+
+    # Get basic configuration
```

Comment:
Since this is the only comment in this function (i.e., since there is no subsequent comment which says, "OK, now put the configuration to use!"), I would omit this -- it doesn't really add enough.

Reply:
At one point, I thought there were more. Hmm.